packages/expo-permissions/ios/EXPermissions/EXPermissions.m

```diff
@@ -28,266 +29,256 @@ @implementation EXPermissions
+- (BOOL)hasGrantedPermissionUsingRequesterClass:(Class)requesterClass
 {
-  NSDictionary *permissions = [self getPermissionsForResource:permissionType];
-
+  NSDictionary *permissions = [self getPermissionUsingRequesterClass:requesterClass];
   if (!permissions) {
-    UMLogWarn(@"Permission with type '%@' not found.", permissionType);
+    UMLogWarn(@"Permission requester '%@' not found.", NSStringFromClass(requesterClass));
     return false;
   }
   
   return [permissions[@"status"] isEqualToString:@"granted"];
 }
 
-- (void)askForPermission:(NSString *)permissionType
-              withResult:(void (^)(NSDictionary *))onResult
-            withRejecter:(UMPromiseRejectBlock)reject
+// From UMPermissionsInterface - use to check permission from code
+// shorthand method that checks both global and per-experience permission
+- (BOOL)hasGrantedPermission:(NSString *)permissionType
 {
-  return [self askForPermissions:@[permissionType]
-                     withResults:^(NSArray<NSDictionary *> *results) {
-                       onResult(results[0]);
-                     }
-                    withRejecter:reject];
+  NSDictionary *permissions = [self getPermissionsForResource:permissionType];
+  if (!permissions) {
+    UMLogWarn(@"Permission with type '%@' not found.", permissionType);
+    return false;
+  }
+  
+  return [permissions[@"status"] isEqualToString:@"granted"];
 }
 
-- (void)askForPermissions:(NSArray<NSString *> *)permissionsTypes
-              withResults:(void (^)(NSArray<NSDictionary *> *))onResults
-             withRejecter:(UMPromiseRejectBlock)reject
+- (void)askForPermissionUsingRequesterClass:(Class)requesterClass
+                                 withResult:(void (^)(NSDictionary *))onResult
+                               withRejecter:(UMPromiseRejectBlock)reject
 {
-  return [self askForPermissionsWithTypes:permissionsTypes
-                              withResults:^(NSDictionary *results) {
-                                NSMutableArray<NSDictionary *> *finalResults = [NSMutableArray new];
-
-                                [permissionsTypes enumerateObjectsUsingBlock:^(NSString * _Nonnull permissionType, NSUInteger idx, BOOL * _Nonnull stop) {
-                                  NSDictionary *result = results[permissionType];
-                                  [finalResults addObject:result];
-                                }];
-                                onResults(finalResults);
-                              }
-                             withRejecter:reject];
+  NSMutableDictionary *permission = [[self getPermissionUsingRequesterClass:requesterClass] mutableCopy];
+  
+  // permission type not found - reject immediately
+  if (permission == nil) {
+    return reject(@"E_PERMISSIONS_UNKNOWN", [NSString stringWithFormat:@"Unrecognized requester: %@", NSStringFromClass(requesterClass)], nil);
+  }
+  
+  BOOL isGranted = [EXPermissions statusForPermission:permission] == UMPermissionStatusGranted;
```

Comment:
What about permissions returning composite results that not necessarily are explicitly granted?
E.g. `NOTIFICATIONS` right now comes with this scenario.